aten/src/ATen/cuda/CublasHandlePool.cpp

```diff
@@ -21,23 +21,28 @@ void destroyCublasHandle(cublasHandle_t handle) {
 #endif
 }
 
-auto pool = std::make_shared<DeviceThreadHandlePool<cublasHandle_t, createCublasHandle, destroyCublasHandle>>();
+using CuBlasPoolType = DeviceThreadHandlePool<cublasHandle_t, createCublasHandle, destroyCublasHandle>;
 
-// Thread local PoolWindows are wrapped by unique_ptrs and lazily-initialized
-// to avoid initialization issues that caused hangs on Windows.
-// See: https://github.com/pytorch/pytorch/pull/22405
-// This thread local unique_ptrs will be destroyed when the thread terminates,
-// releasing its reserved handles back to the pool.
-thread_local std::unique_ptr<decltype(pool)::element_type::PoolWindow> myPoolWindow;
+CuBlasPoolType &getPool()
+{
+  static auto pool = std::make_shared<CuBlasPoolType>();
+  return *pool;
+}
 
 } // namespace
 
 cublasHandle_t getCurrentCUDABlasHandle() {
   int device;
   AT_CUDA_CHECK(cudaGetDevice(&device));
 
-  if (!myPoolWindow)
-    myPoolWindow.reset(pool->newPoolWindow());
+  // Thread local PoolWindows are wrapped by unique_ptrs and lazily-initialized
+  // to avoid initialization issues that caused hangs on Windows.
+  // See: https://github.com/pytorch/pytorch/pull/22405
+  // This thread local unique_ptrs will be destroyed when the thread terminates,
+  // releasing its reserved handles back to the pool.
+  thread_local std::unique_ptr<CuBlasPoolType::PoolWindow> myPoolWindow(
```

Comment:
per my [first comment](https://github.com/pytorch/pytorch/pull/39345#issuecomment-643593751), I don't think moving `myPoolWindow`s from a file scope thread local to a function scope thread local actually helps enforce 1->2 ordering, but per my [second comment](https://github.com/pytorch/pytorch/pull/39345#issuecomment-643601421) in which I came to my senses, now that `PoolWindow`s hold `weak_ptr`s to their `DeviceThreadHandlePool`s, I don't think we care about 1->2 ordering at all!

So i think it's fine to construct `myPoolWindow`s in `getCurrent*Handle`s, but I also think it would be fine to leave them as file scope thread locals.  whatever you prefer.